.github/workflows/workflow.yml

```diff
@@ -21,9 +21,67 @@ jobs:
     - name: Run Tox Check
       run: tox -e check
 
-  build:
+  data-cache:
     needs: check
     runs-on: ubuntu-latest
+    outputs:
+      cache_hash: ${{ steps.compute-cache-hash.outputs.cache_hash }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.9'
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -e ".[dev]"
+      - name: Restore Cache
+        uses: actions/cache@v4
+        with:
+          key: cached-data-${{ hashFiles('**/registry.py', '**/*workflow.yml') }}
+          restore-keys: |
+            cached-data-
```

Comment:
Do we still need this?

Reply:
It may not be necessary, but in the event that the cache key is changed this would access the old cache and potentially save us downloading files stored there.

Reply:
discussed.... just confirm its (pooch) not going to error on registry update. Thanks

Reply:
I've confirmed that in the case where a file's hash does not match the hash in the registry pooch will not error, but update the file.

Reply:
Great! Thanks!